src/librustc/ich/impls_mir.rs

```diff
@@ -20,7 +20,8 @@ use std::mem;
 impl_stable_hash_for!(struct mir::GeneratorLayout<'tcx> { fields });
 impl_stable_hash_for!(struct mir::SourceInfo { span, scope });
 impl_stable_hash_for!(enum mir::Mutability { Mut, Not });
-impl_stable_hash_for!(enum mir::BorrowKind { Shared, Unique, Mut });
+impl_stable_hash_for!(struct mir::BorrowKind { mut_kind, introduced_via_autoref });
```

Comment:
bikeshed: I would prefer the "operational" `always_allow_2phi` to the descriptive `introduced_via_autoref`.

Reply:
please call it "two phase", but yes that's a good point. Operational is often better.

Reply:
okay, this distinction doesn't matter to me, so I'll make this change.